### ISSUE TITLE
Is there a way to get the coordinates of a tap event on a page?

### ISSUE THREAD

**Issue by GP4cK** [endigo/flutter_pdfview#113]
I would like to let the user add markers on the PDF. For that, I would need 2 things:
1. Being able to listen to tap events on the PDF to get a set of coordinates X and Y
2. Give the viewer a list of coordinates and some colors so that it can place the markers on top of the document

I understand that it may not be supported at the moment but may I have some indications on how to start if I had to implement this?
Thanks a lot!

**Comment by itmediacreations** [comment]
Have you tried to spread out GestureDetector in Stack with PDF like this:
```
Stack(
  children: [
     GestureDetector(
      onLongPressStart: (touchDetails) {
        double x = touchDetails.globalPosition.dx; // touchDetails.localPosition
        // do something with coordinates
      },
    ),
    PDF(),
  ],
),`

**Comment by GP4cK** [comment]
Yes, the problem is when you zoom and pan, you don't get the right coordinates...

**Comment by GP4cK** [comment]
I've tried to pass `ScaleGestureRecognizer` and `PanGestureRecognizer` as `gestureRecognizers` but they don't seem to catch any gesture.
I've asked a question on [SO](https://stackoverflow.com/q/64475004/3466729) as this is not entirely related to the viewer itself but more to AndroidView and UiKitView

**Comment by sammusaev** [comment]
@GP4cK any luck? I'm in a similar situation right now, and am considering disabling zoom via some overlay, and having controls for page scrolling be done via buttons.

**Comment by GP4cK** [comment]
@sammusaev I ended up converting the PDF to png in the backend and use an [InteractiveViewer](https://api.flutter.dev/flutter/widgets/InteractiveViewer-class.html).

**Comment by chrisbinsunny** [comment]
@sammusaev  @PcolBP 

I found a solution for the issue. I was checking for the presense of onTapDown and added 

    gestureRecognizers: Set()
            ..add(Factory<TapGestureRecognizer>(() => TapGestureRecognizer()
              ..onTapDown = (tap) {
                print("hell bro");
              })),

But, it didn't work somehow. I was checking for other solutions and thought to add GestureDetector and detect tap from that.

So, I just wrapped PDFView with a GestureDetector. Surprisingly, gestureRecognizers inside PDFView worked after adding GestureDetector() Widget. Since I was checking for onTapDown event with gestureRecognizers, I added onTap parameter in GestureDetector. 

My Widget looked like this:

    GestureDetector(
        onTap: (){}, //This has to be added. Won't work without this.
        child: PDFView(
          filePath: widget.link,
          onError: (error) {
            print(error.toString());
          },
          onPageError: (page, error) {
            print('$page: ${error.toString()}');
          },
          onViewCreated: (PDFViewController pdfViewController) {
            pdfController=pdfViewController;
          },
          onPageChanged: (a, b){
            visible=false;
          },
          gestureRecognizers: Set()
            ..add(Factory<TapGestureRecognizer>(() => TapGestureRecognizer()
              ..onTapDown = (tap) {
                print("Tap Down Gesture Detected");
              })),
        ),
      )

So, for your case I guess you'll have to add the following code:

    return GestureDetector(
                onScaleStart: (a){}, //This has to be added. Won't work without this.
                onPanStart: (a){}, //This has to be added. Won't work without this.
                child: PDFView(
                  filePath: filePath,
                  gestureRecognizers: Set()
                    ..add(
                        Factory<OneSequenceGestureRecognizer>(() => scale)
                    )
                    ..add(
                    Factory<OneSequenceGestureRecognizer>(() => pan),
                ),)
              );

@GP4cK I added the solution in your [SO](https://stackoverflow.com/a/75001577/11546127) question. Please check that too...
Hope that helps...
